src/printer.js

```diff
@@ -989,6 +989,7 @@ function genericPrintNoParens(path, options, print) {
     case "ForStatement": {
       const body = adjustClause(path.call(print, "body"), options);
 
+      // We want to keep dangling comments above the loop to stay consistent.
```

Comment:
I think explaining what comment is dangling is helpful (I'm not even sure why this comment was classified as dangling). Can you say something about the comment being between the `for` statement and the parentheses?